board/app/models/comment.php

```diff
@@ -1,26 +1,186 @@
 <?php
 
-//Comment Validation
-class Comment extends AppModel
-{
-  const MIN_USERNAME_LENGTH = 1;
-  const MIN_BODY_LENGTH = 1;
+class Comment extends AppModel {
 
-  const MAX_USERNAME_LENGTH = 20;
-  const MAX_BODY_LENGTH = 200;
+//Comment Length Validation
+const MIN_USERNAME_LENGTH = 1;
+const MIN_BODY_LENGTH = 1;
 
-  public $validation = array(
+const MAX_USERNAME_LENGTH = 20;
+const MAX_BODY_LENGTH = 140;
+
+public $validation = array(
     "username" => array(
-      "length" => array(
-        "validate_between", self::MIN_USERNAME_LENGTH, self::MAX_USERNAME_LENGTH,
-      ),
-    ),
+        "length" => array(
+            "validate_between", self::MIN_USERNAME_LENGTH, self::MAX_USERNAME_LENGTH,
+            ),
+        ),
 
     "body" => array(
-      "length" => array(
-        "validate_between", self::MIN_BODY_LENGTH, self::MAX_BODY_LENGTH,
-      ),
-    ),
-  );
-}
-?> 
+        "length" => array(
+            "validate_between", self::MIN_BODY_LENGTH, self::MAX_BODY_LENGTH,
+            ),
+        ),
+    );
+
+    public static function countAll($thread_id) {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment 
+            WHERE thread_id = ? ", array($thread_id));
+    }
+
+    public static function getAll($offset, $limit, $thread_id) {
+        $comments = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM comment
+                WHERE thread_id = ?
+                    ORDER BY created ASC LIMIT {$offset}, {$limit}", array($thread_id));
+        
+        foreach ($rows as $row) {
+            $comments[] = new self($row);
+            }
+        return $comments;
+        }
+
+    public function write(Comment $comment, $thread_id) {
+    
+    if(!$comment->validate()) {
+        throw new ValidationException('invalid comment');
+    }
```

Comment:
based from this [comment] (https://github.com/maragatchalian/Board/pull/2/files#r27275107), please remove $comment as parameter. You may use $this->validate() instead of $comment->validate() in L47